modules/ROOT/pages/dataweave-selectors.adoc

```diff
@@ -48,6 +41,618 @@ xref:dataweave-cookbook-extract-data#selector_metadata[Extract Data].
+ns html http://www.w3.org/TR/html4/
+ns furniture https://www.w3schools.com/furniture
+---
+root: {
+     a: payload.root.html#table,
+     b: payload.root.furniture#table
+ }
+----
+
+==== Output
+
+The script outputs children of the `h:table` element under element `a` and
+the children of the `f:table` under element `b`.
+
+[source,xml, linenums]
+----
+<?xml version='1.0' encoding='UTF-8'?>
+<root>
+  <a>
+    <h:tr xmlns:h="http://www.w3.org/TR/html4/">
+      <h:td>Apples</h:td>
+      <h:td>Bananas</h:td>
+    </h:tr>
+  </a>
+  <b>
+    <f:tr xmlns:f="https://www.w3schools.com/furniture">
+      <f:name>African Coffee Table</f:name>
+      <f:width>80</f:width>
+      <f:length>120</f:length>
+    </f:tr>
+  </b>
+</root>
+----
+
+=== Example
+
+This example selects XML values using a local name `table` from the `h:table`
+element.
+
+==== Input
+
+The input payload contains XML with a `table` element that contains
+the namespace `"http://www.w3.org/TR/html4/"`.
+
+[source,xml, linenums]
+----
+<root>
+  <h:table xmlns:h="http://www.w3.org/TR/html4/">
+    <h:tr>
+      <h:td>Apples</h:td>
+      <h:td>Bananas</h:td>
+    </h:tr>
+  </h:table>
+</root>
+----
+
+==== Source
+
+The DataWeave script selects values of children in the `table` element
+by using a local name of the XML element `table` instead of its namespace.
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+output application/xml
+---
+root: { a: payload.root.table }
+----
+
+==== Output
+
+The script outputs XML that contains the values of children in the
+`table` element.
+
+[source,xml, linenums]
+----
+<?xml version='1.0' encoding='UTF-8'?>
+<root>
+  <a>
+    <h:tr xmlns:h="http://www.w3.org/TR/html4/">
+      <h:td>Apples</h:td>
+      <h:td>Bananas</h:td>
+    </h:tr>
+  </a>
+</root>
+----
+
+[[attribute_selector]]
+== Attribute Selector
+
+The attribute selector returns the first attribute value that matches the
+selected name expression. If no key matches, the selection returns the value
+`null`.
+
+The following example selects the value of an XML attribute.
+
+=== Input
+
+The input payload is an XML `user` element that contains the attribute
+`name="Weave"`.
+
+[source,xml, linenums]
+----
+<user name="Weave"/>
+----
+
+=== Source
+
+The DataWeave script selects the value of the `name` attribute
+from the input payload (`<user name="Weave"/>`). Notice that the `@`
+indicates an attribute selection.
+
+[source,DataWeave, linenums]
+----
+%dw 2.0
+output application/json
+---
+payload.user.@name
+----
+
+=== Output
+
+The DataWeave script outputs the value of `name` attribute.
+
+[source,json,linenums]
+----
+"Weave"
+----
+
+[[multi_value_selector]]
+== Multi-Value Selector
+
+Instead of returning the value of the first matching key in an array
+of objects, the multi-value selector (`*`) returns an array containing
+values to the matching key (for example, `*user` where `user` is
+the key). The selector does not return the values of descendants,
```

Comment:
Should we use "children" here, to be consistent with previous text?
```suggestion
the key). The selector does not return the values of children; it returns  
```

Reply:
It's all descendants, not just the children. Includes grandchildren, etc.

Reply:
replied on slack